scripts/ci/hermetic-network-guard.c

```
@@ -0,0 +1,397 @@
+DYLD_INTERPOSE(guarded_write, write);
+DYLD_INTERPOSE(guarded_writev, writev);
+#else
+int connect(int socket_fd, const struct sockaddr *address, socklen_t address_length) {
+    return guarded_connect(socket_fd, address, address_length);
+}
+
+ssize_t send(int socket_fd, const void *buffer, size_t length, int flags) {
+    return guarded_send(socket_fd, buffer, length, flags);
+}
+
+int sendmmsg(
```

Cppcheck (2.21.0)

[style] The function 'sendmmsg' is never used.

(unusedFunction)